src/script/modules/peeking-indicator/index.ts

```diff
@@ -0,0 +1,302 @@
+import settings from '../../lib/settings';
+import Module from '../../lib/module';
+import { getSnapchatStore } from '../../utils/snapchat';
+import styles from './index.module.css';
+import { logInfo } from '../../lib/debug';
+
+const store = getSnapchatStore();
+let initialized = false;
+
+class PeekingIndicator extends Module {
+  constructor() {
+    super('Peeking Indicator');
+    settings.on('HALF_SWIPE_NOTIFICATION.setting:update', this.load.bind(this));
+
+    this.load();
+  }
+
+  findSnapchatConversationContainers() {
+    const possibleContainers = [
+      ...document.querySelectorAll('.O4POs'),
+      ...document.querySelectorAll('[data-testid="conversation_item_container"]'),
+      ...document.querySelectorAll('[data-role="conversation_item"]'),
+      ...document.querySelectorAll('[role="listitem"]'),
+    ];
+    return possibleContainers;
+  }
+
+  initMutationObserver() {
+    if (initialized) return;
+
+    initialized = true;
+
+    const observer = new MutationObserver((mutations) => {
+      let shouldCheckForContainers = false;
+
+      mutations.forEach((mutation) => {
+        if (mutation.type === 'childList' && mutation.addedNodes.length > 0) {
+          shouldCheckForContainers = true;
+        }
+      });
+
+      if (shouldCheckForContainers) {
+        this.processConversationContainers();
+      }
+    });
+
+    observer.observe(document.body, {
+      childList: true,
+      subtree: true,
+    });
+
+    this.processConversationContainers();
+    setInterval(() => this.processConversationContainers(), 2000);
+  }
+
+  processConversationContainers() {
+    const enabled = settings.getSetting('HALF_SWIPE_NOTIFICATION');
+    if (!enabled) return;
+
+    const containers = this.findSnapchatConversationContainers();
+    containers.forEach((container) => this.addPeekingIndicator(container));
+  }
+
+  addPeekingIndicator(container: Element) {
+    if (container.querySelector(`.${styles.peeking}`)) {
+      return;
+    }
+
+    const peekingDiv = document.createElement('div');
+    peekingDiv.className = styles.peeking;
+
+    const peekingItem = document.createElement('div');
+    peekingItem.className = styles.peekingItem;
+
+    peekingDiv.appendChild(peekingItem);
+    container.prepend(peekingDiv);
+  }
+
+  getConversationIdFromElement(container: Element): string | null {
```

Comment:
There's probably a more reliable way for this, by using exposed react props instead of searching DOM attributes